riscv.c

```diff
@@ -758,47 +758,124 @@ static bool op_amo(struct riscv_t *rv, uint32_t inst)
+#endif
+
+    // clang-format off
+    TABLE_TYPE jump_table[] = {
+    //  000         001           010        011           100         101        110   111
+        OP(load),   OP(load_fp),  OP(unimp), OP(misc_mem), OP(op_imm), OP(auipc), OP(unimp), OP(unimp), // 00
+        OP(store),  OP(store_fp), OP(unimp), OP(amo),      OP(op),     OP(lui),   OP(unimp), OP(unimp), // 01
+        OP(madd),   OP(msub),     OP(nmsub), OP(nmadd),    OP(fp),     OP(unimp), OP(unimp), OP(unimp), // 10
+        OP(branch), OP(jalr),     OP(unimp), OP(jal),      OP(system), OP(unimp), OP(unimp), OP(unimp), // 11
+    };
+    // clang-format on
+
+#ifdef ENABLE_COMPUTED_GOTO
+#define DISPATCH()                                      \
+    {                                                   \
+        if (rv->csr_cycle >= cycles_target || rv->halt) \
+            goto exit;                                  \
+        else {                                          \
+            /* fetch the next instruction */            \
+            inst = rv->io.mem_ifetch(rv, rv->PC);       \
+            /* standard uncompressed instruction */     \
+            if ((inst & 3) == 3) {                      \
+                index = (inst & INST_6_2) >> 2;         \
+                goto *jump_table[index];                \
+            } else {                                    \
+                /* TODO: compressed instruction*/       \
+                assert(!"Unreachable");                 \
+            }                                           \
+        }                                               \
+    }
+
+#define EXEC(instr)                   \
+    {                                 \
+        /* dispatch this opcode */    \
+        if (!op_##instr(rv, inst))    \
+            goto exit;                \
+        /* increment the cycles csr*/ \
+        rv->csr_cycle++;              \
+    }
+
+#define TARGET(instr)         \
+op_##instr :                  \
+    EXEC(instr);              \
+    DISPATCH();
+
+    DISPATCH();
+
+    // main loop
+    TARGET(load)
+    TARGET(op_imm)
+    TARGET(auipc)
+    TARGET(store)
+    TARGET(op)
+    TARGET(lui)
+    TARGET(branch)
+    TARGET(jalr)
+    TARGET(jal)
+    TARGET(system)
+#ifdef ENABLE_Zifencei
+    TARGET(misc_mem)
+#endif
+#ifdef ENABLE_RV32A
+    TARGET(amo)
+#endif
+    TARGET(unimp)
+
+exit:
+    return;
 
+#undef DISPATCH
+#undef EXEC
+#undef TARGET
+#else   // ENABLE_COMPUTED_GOTO = 0
     while (rv->csr_cycle < cycles_target && !rv->halt) {
         // fetch the next instruction
-        const uint32_t inst = rv->io.mem_ifetch(rv, rv->PC);
+        inst = rv->io.mem_ifetch(rv, rv->PC);
 
         // standard uncompressed instruction
         if ((inst & 3) == 3) {
-            const uint32_t index = (inst & INST_6_2) >> 2;
+            index = (inst & INST_6_2) >> 2;
 
-	    // dispatch this opcode
-            const opcode_t op = opcodes[index];
+            // dispatch this opcode
```

Comment:
Use `clang-format -i` to ensure the consistent style.

Reply:
+ I used clang-format but it doesn't complain about this line. 
+ It did complain about nested #if at [line 790](https://github.com/sysprog21/rv32emu-next/pull/2/files/7087c7f3ecdd630ef92198109b655dab816d50e6#diff-76a818eab913b98b885f954273a467dea476f7ddbfd5886f97b1e56cb0e2b58dR790-R796). However, I think it looks better with indentation.
+ It also tried to make [the label](https://github.com/sysprog21/rv32emu-next/pull/2/files/7087c7f3ecdd630ef92198109b655dab816d50e6#diff-76a818eab913b98b885f954273a467dea476f7ddbfd5886f97b1e56cb0e2b58dR837) and `EXEC()` into single line but it looks better separated for me.

Should I just use clang-format or leave it as is ?

Reply:
Use `clang-format off` and `clang-format on` to switch manual indentation.